### PR TITLE
Fix/move to elastic enterprise search client

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

Use `elastic-enterprise-search` client library  when connecting to Elastic Enterprise Search, keeps the deprecated `elastic-app-search` to connect to Swiftype AppSearch

## What does this PR do?

This PR use the  `elastic-enterprise-search` next to `elastic-app-search` client library and select which one to use basing on options used.
Rename some references of App Search to Enterprise Search.
Due to the depedency of  `logstash-core ~>7.13.0` on `elasticsearch ~>5` this PR is could be merged/published after Logstash `7.14` is released, because it contains the fix https://github.com/elastic/logstash/pull/12874

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files (and/or docker env variables)
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [x] test against Swiftype SaaS account
- [x] test against Elastic Enterprise Search SaaS account
- [x] test against Elastic Enterprise Search on-prem instance

## Related issues

- #13

### CODE REVIEW

**Comment by andsel** [comment]
The CI problems:
- on `7.x` is due to the fact that `logstash-core` up to version `< 7.14` requires `elastic-search ~> 5` fixed with https://github.com/elastic/logstash/pull/12874 on `7.x` branch
- on `8.0` is due to the fact that AppSearch 8.0 is an alias for latest `7.x` (currently 7.13) and doesn't yet support Elasticsearch 8.0

**Comment by andsel** [comment]
The red on CI is related to recent Workplace Search updates on `8.x` `snapshot` versions.
Unrelated to this PR

**Comment by andsel** [comment]
Red CI is tracked in #14

File: docs/output-elastic_app_search.asciidoc [hunk 1]
```diff
@@ -131,16 +131,43 @@ output {
...
 [id="plugins-{type}s-{plugin}-path"]
 ===== `path`
 
 * Value type is <<string,string>>
 * Default value is `/api/as/v1/`
 
 The path that is appended to the `url` parameter when connecting to a https://www.elastic.co/downloads/app-search[self-managed Elastic App Search].
+The `path` is used only for Swiftype App Search API, if configured it needs the `host` refers to a Swiftype's AppSearch instance.
+
+NOTE: This is valid only for Swiftype SaaS AppSearch accounts, could be removed in future versions.
```
Comment (jsvd) [inline docs/output-elastic_app_search.asciidoc:159]: Should it be clearer that this is a setting for the deprecated service and doesn't need tweaking for the Elastic Service?

```suggestion

The `path` setting is used only for Swiftype SaaS AppSearch and could be removed in future versions.
If configured, it needs the `host` which refers to a Swiftype's AppSearch instance.

Note: When connecting to a https://www.elastic.co/downloads/app-search[self-managed Elastic App Search] the path is appended to the `url` parameter.
```

File: logstash-integration-elastic_enterprise_search.gemspec [hunk 2]
```diff
@@ -27,6 +27,8 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency "logstash-core-plugin-api", "~> 2.0"
   s.add_runtime_dependency "logstash-codec-plain"
   s.add_runtime_dependency "elastic-app-search", '~>7.8.0'
+  s.add_runtime_dependency "elastic-enterprise-search", '~>7.15.1'
```
Comment (jsvd) [inline logstash-integration-elastic_enterprise_search.gemspec:30]:
```suggestion
  s.add_runtime_dependency "elastic-enterprise-search", '~>7.16.0'
```

**Review by jsvd — COMMENTED** [review]
Overall LGTM but we can improve docs to better differentiate the settings needed for either the Swiftype or Elastic services.

I successfully tested this PR on:
1. a local deployment of AppSearch ✅ 
2. a Swiftype AppSearch Saas account ✅ 
3. an Elastic AppSearch SaaS account ✅

File: docs/output-elastic_app_search.asciidoc [hunk 3]
```diff
@@ -131,16 +131,43 @@ output {
 * Value type is <<string,string>>
 * There is no default value
 
-The hostname of the App Search API that is associated with your App Search account.
+The hostname of the App Search SaaS API that is associated with your App Search account.
+This is used only by Swiftype App Search, and has the form of `host-1ab2cd`.
 Set this when using the https://www.elastic.co/cloud/app-search-service[Elastic App Search managed service].
 
+NOTE: This is valid only for Swiftype SaaS AppSearch accounts, could be removed in future versions.
```
[the remark this answers is not in the document]
Comment (andsel) [reply docs/output-elastic_app_search.asciidoc:138]: Do you mean we should point to https://swiftype.com/documentation/app-search/getting-started ?

[the remark this answers is not in the document]
Comment (andsel) [reply docs/output-elastic_app_search.asciidoc:138]:
```suggestion
This is used only by Swiftype App Search, and has the form of `host-1ab2cd`.
Set this when using the https://swiftype.com/documentation/app-search/getting-started [Swiftype App Search managed service].

NOTE: This is valid only for Swiftype SaaS AppSearch accounts, could be removed in future versions.
```

[the remark this answers is not in the document]
Comment (jsvd) [reply docs/output-elastic_app_search.asciidoc:138]: Correct, thanks

**Pull request merged**